extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -104,17 +106,68 @@ export class DeployDatabaseDialog {
 		});
 	}
 
-	private async deploy(): Promise<void> {
-		// TODO: hook up with build and deploy
+	public async getConnectionUri(): Promise<string> {
 		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+		let connId: string;
+
+		if (this.dataSourcesRadioButton?.checked) {
+			const dataSource = (this.dataSourcesDropDown!.value! as DataSourceDropdownValue).dataSource;
+
+			const connProfile: azdata.IConnectionProfile = {
+				serverName: dataSource.server,
+				databaseName: dataSource.database,
+				connectionName: dataSource.name,
+				userName: '',
+				password: '', // TODO: secure password storage
```

Comment:
add to datasources.json, and add a TODO note there about storing/accessing/prompting for creds